src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -66,12 +66,12 @@ export class SelectBox extends vsSelectBox {
 
 	constructor(options: SelectOptionItemSQL[] | string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
 		let optionItems: SelectOptionItemSQL[];
-		if (Array.isArray<string>(options)) {
-			optionItems = (options as string[]).map(o => {
+		if (typeof (options[0]) === 'string' && Array.isArray<string>(options)) {
```

Comment:
Just to be safe should check options.length > 0 too

Reply:
I can probably just flip the conditions.

Reply:
[] is an array, but your check will throw exception here.

Reply:
That's true, although I'm going to end up just throwing another error to the user here.